webhook/webhook.go

```diff
@@ -0,0 +1,104 @@
+package webhook
+
+import (
+  "crypto/hmac"
+  "crypto/sha256"
+  "encoding/hex"
+  "net/http"
+  "fmt"
+  "errors"
+  "encoding/json"
+  "bytes"
+  "encoding/gob"
+
+  log "github.com/gophish/gophish/logger"
+)
+
+const (
+  DefaultTimeoutSeconds = 10
+  MinHttpStatusErrorCode = 400
+  SignatureHeader = "X-Gophish-Signature"
+)
+
+
+//TODO
+
+type Sender interface {
+  Send(url string, secret string, data interface{}) error
+}
+
+type defaultSender struct {
+  client *http.Client
+} 
+
+var senderInstance Sender
```

Comment:
Instead of the pattern where you check if `senderInstance` is `nil` then call `newDefaultSender`, you might instead just instantiate it here. You can see an example in the [`net/http` package](https://golang.org/src/net/http/client.go#L107). This would look something like:

```
var senderInstance = &defaultSender{
    client: &http.Client{
        Timeout: DefaultTimeoutSeconds,
    }
}
```

Then you can remove `newDefaultSender` and all the logic in the functions below.

Reply:
Done